### DISCUSSION TITLE
How to implement state machine for Async operations?

### DISCUSSION THREAD

**Discussion by makeevvd** [gcanti/fp-ts#1832] [General]
Either gives us an option to process Error state and Success state, but for frontend rendering I need also a pending state of asynchronous functions. What is the best way (monad) to implement a state machine, which will give us Error, Pending and Success state, so I can fold 3 values, depending on whether my computation is in the progress, resolved or rejected.

**Comment by samhh** [comment]
[`RemoteData`](https://github.com/devexperts/remote-data-ts)

**Comment by 0x706b** [comment]
Or [Datum](https://github.com/baetheus/datum) (includes a Reloading state)
